simpleflow/download.py

```diff
@@ -0,0 +1,82 @@
+import errno
+import functools
+import hashlib
+import logging
+import os
+
+from simpleflow.storage import pull
+
+
+LOCAL_BIN_DIR = "/tmp/simpleflow-binaries"
+
+
+logger = logging.getLogger(__name__)
+
+
+class RemoteBinary(object):
+    """
+    This class identifies if binary is available in $PATH so it can be used.
+    If the binary is not available, simpleflow will try to download it and
+    put it in a dedicated folder so it can be used. It will also prepend this
+    folder to $PATH before forking to the real activity worker.
+    """
+    def __init__(self, name, remote_location):
+        """
+        :param name: name of the binary to be downloaded
+        :type  name: str
+        :param remote_location: remote location where to download the binary from (only S3 for now)
+        :type  remote_location: str
+        """
+        self.name = name
+
+        # limit ourselves to S3 for now
+        assert remote_location.startswith("s3://")
```

Comment:
Note: ignored with `python -O` :stuck_out_tongue:

Reply:
Didn't know! What's the best way to add quick assertions in production code, should we keep it like this?

Reply:
I guess it's to remove the first `if` from
```python
if __debug__:
    if not expression: raise AssertionError
```
?

Reply:
No, it's just that I saw it here and there and I thought it was the correct+easy way to crash at runtime if something goes really wrong (6 occurrences in simpleflow, 99 in our private codebase)

Reply:
It's for development only, as in C :slightly_smiling_face: (except that nobody uses `python -O` anyway)